### ISSUE TITLE
encoding/xml: missing nested namespace not handled

### ISSUE THREAD

**Issue by anacrolix** [golang/go#7113]
<pre>Nested XML namespaces aren't marshalled correctly. Setting an element namespace at the
default level as Go does, implicitly sets nested elements to the same namespace if they
don't override it.

&lt;a xmlns=&quot;b&quot;&gt;&lt;c/&gt;&lt;/a&gt;

The namespace of c, is b. However marshalling from Go data structures in which the
object corresponding to c above has no namespace set should give

&lt;a xmlns=&quot;b&quot;&gt;&lt;c xmlns=&quot;&quot;/&gt;&lt;/a&gt;

The usual way to deal with this is to use namespace prefixes, but I don't think that's a
requirement.</pre>

**Comment by jordan2175** [comment]
Please see my example and scenario on issue #9527 as well.   Also, there is another issue with prefixes that is going to be highly related to this #9519 / #6800.  I would love to see both of these fixed in 1.5 if I can beg enough.

**Comment by lucasdss** [comment]
I'm writing an RPM manager using GO and I ended up  with this problem too. The XML file primary.xml provided by any RPM repository uses "rpm:entry" and others tags with the "rpm" as namespace prefix. It took me two days until I figured out It's a bug. Well, I did some string.replace and it works, but not as good as xml.Marshal.

**Comment by rsc** [comment]
See #11841.

**Comment by rsc** [comment]
Blocked on #13400.

**Comment by snurmine** [comment]
Can I help with issue ?
